python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+           "BfKernelMock", "DefectMock", "CrosstalkCoeffMock", "TransmissionMock",
+           "DataRefMock"]
+
+
+class IsrMockConfig(pexConfig.Config):
+    r"""Configuration parameters for isrMock.
+
+    These parameters produce generic fixed position signals from
+    various sources, and combine them in a way that matches how those
+    signals are combined to create real data.  The camera used is the
+    test camera defined by the afwUtils code.
+    """
+    # Detector parameters.  "Exposure" parameters.
+    isLsstLike = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="If True, products have one raw image per amplifier, otherwise, one raw image per detector",
+    )
+    isTrimmed = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="If True, amplifiers have been trimmed and mosaicked to remove regions outside the data BBox",
+    )
+    detectorIndex = pexConfig.Field(
+        dtype=int,
+        default=20,
+        doc="Index for the detector to use.  The default value uses a standard 2x4 array of amps.",
+    )
+    rngSeed = pexConfig.Field(
+        dtype=int,
+        default=20000913,
+        doc="Seed for random number generator used to add noise.",
+    )
+    gain = pexConfig.Field(
+        dtype=float,
+        default=1.0,
+        doc="Gain for simulated data.",
+    )
+    expTime = pexConfig.Field(
+        dtype=float,
+        default=5.0,
+        doc="Exposure time for simulated data.",
+    )
+
+    # Signal parameters
+    skyLevel = pexConfig.Field(
+        dtype=float,
+        default=1000.0,
+        doc="Background contribution to be generated from 'the sky'.",
+    )
+    skySigma = pexConfig.Field(
+        dtype=float,
+        default=32.0,
+        doc="Background noise contribution to be generated from 'the sky'.",
+    )
+    sourceFlux = pexConfig.Field(
+        dtype=float,
+        default=45000.0,
+        doc="Flux level (in DN) of a simulated 'astronomical source'.",
+    )
+    overscanScale = pexConfig.Field(
+        dtype=float,
+        default=100.0,
+        doc="Amplitude of the ramp function to add to overscan data.",
+    )
+    biasLevel = pexConfig.Field(
+        dtype=float,
+        default=8000.0,
+        doc="Background contribution to be generated from the bias offset.",
+    )
+    biasSigma = pexConfig.Field(
+        dtype=float,
+        default=90.0,
+        doc="Background noise contribution to be generated from the bias.",
+    )
+    darkRate = pexConfig.Field(
+        dtype=float,
+        default=5.0,
+        doc="Background level contribution to be generated from dark current.",
+    )
+    darkTime = pexConfig.Field(
+        dtype=float,
+        default=5.0,
+        doc="Exposure time for the dark current contribution.",
+    )
+    flatScale = pexConfig.Field(
+        dtype=float,
+        default=320.0,
+        doc="Scale factor to apply to the flat dome.  Should not need to be so big.",
```

Comment:
This was unclear, so I've switched to a hopefully more intuitive configuration option (flatDrop) with better description (Fractional flux drop due to flat from center to edge of detector along x-axis.), and updated the flat method to convert the option into the necessary math.

Reply:
That sounds great. I'm a bit fan of really verbose names, so feel free to ignore this, but I'd probably call it something like `flatFieldDropOff` rather than `flatDrop`, but like I said, your call.